### PR TITLE
LF: fix archive decoder for GREATER_DATE

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala [hunk 1]
```diff
@@ -714,9 +714,9 @@ class DecodeV1Spec
 
     s"reject comparison builtins as is if version >= ${LV.Features.genComparison}" in {
 
-      forEveryVersionSuchThat(_ < LV.Features.genComparison) { version =>
+      forEveryVersionSuchThat(_ >= LV.Features.genComparison) { version =>
```
Comment (cocreature) [inline daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:717]: :facepalm:

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala [hunk 2]
```diff
@@ -714,9 +714,9 @@ class DecodeV1Spec
...
         val decoder = moduleDecoder(version)
-        forEvery(genericComparisonBuiltinCases) { (proto, _) =>
+        forEvery(comparisonBuiltinCases) { (proto, _) =>
```
Comment (cocreature) [inline daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:719]: I’m confused, where did we rename that? Are we still using `genericComparisonBuiltinCases`?

Comment (remyhaemmerle-da) [reply daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:719]: The test ` s"reject comparison builtins as is if version >= ${LV.Features.genComparison}" `  was just wrong. 
Its body was copied from `"translate generic comparison builtins as is if version < ${LV.Features.genComparison}"` but it was never adapted it. 
You can check the body of those tests were excatly the same before my fix.

Comment (cocreature) [reply daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:719]: ah right, thanks for the explanaiton.

**Review by cocreature — APPROVED** [review]
Good catch!

**Pull request merged**